File: frontend/lib/src/features/videos/models/video_create_request.dart
import 'video_visibility.dart';

class VideoCreateRequest {
  const VideoCreateRequest({
    required this.categoryId,
    required this.caption,
    required this.visibility,
    required this.isPublished,
    this.earlyAccessDays,
  });

  final String categoryId;
  final String caption;
  final VideoVisibility visibility;
  final bool isPublished;
  final int? earlyAccessDays;

  Map<String, String> toFormFields() {
    return {
      'CategoryId': categoryId,
      'Caption': caption,
      'Visibility': visibility.value.toString(),
      'IsPublished': isPublished.toString(),
      if (earlyAccessDays != null)
        'EarlyAccessDays': earlyAccessDays.toString(),
    };
  }
}

File: Backend/src/VidiVideo.Api/Requests/VideoCreateRequest.cs
using VidiVideo.Domain.Enums;

namespace VidiVideo.Api.Requests
{
    public sealed class VideoCreateRequest
    {
        public Guid CategoryId { get; init; }
        public string Caption { get; init; } = string.Empty;
        public VideoVisibility Visibility { get; init; }
        public bool IsPublished { get; init; }
        public int? EarlyAccessDays { get; init; }
        public IFormFile VideoFile { get; init; } = null!;
        public IFormFile ThumbnailFile { get; init; } = null!;
    }
}
